pkg/registration/spoke/registration/hub_accept_controller.go

```diff
@@ -2,44 +2,53 @@ package registration
 
 import (
 	"context"
+	"time"
 
 	"github.com/openshift/library-go/pkg/controller/factory"
 	"github.com/openshift/library-go/pkg/operator/events"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 
+	clientset "open-cluster-management.io/api/client/cluster/clientset/versioned"
 	clusterv1informer "open-cluster-management.io/api/client/cluster/informers/externalversions/cluster/v1"
-	clusterv1listers "open-cluster-management.io/api/client/cluster/listers/cluster/v1"
 )
 
 // hubAcceptController watch ManagedCluster CR on hub after spoke bootstrap done
 // If the managedCluster.Spec.HubAccetpsClient is false, then mark the connection as failed.
 //
 // Note that: when the HubAcceptsClient is false, the clusterrole and clusterrolebinding will be removed.
-// Then the agent will not be able to get or watch the managedCluster CR.
+// Then the agent will not be able to get or watch the managedCluster CR. This is why we need to resync every minute and check if err is forbidden.
 // That means the controller can only handle the case when the HubAcceptsClient change from true to false.
 type hubAcceptController struct {
 	clusterName       string
-	hubClusterLister  clusterv1listers.ManagedClusterLister
+	hubClusterClient  clientset.Interface
 	handleAcceptFalse func(ctx context.Context) error
 	recorder          events.Recorder
 }
 
 func NewHubAcceptController(clusterName string, hubClusterInformer clusterv1informer.ManagedClusterInformer,
+	hubClusterClient clientset.Interface,
 	handleAcceptFalse func(ctx context.Context) error, recorder events.Recorder) factory.Controller {
 	c := &hubAcceptController{
 		clusterName:       clusterName,
-		hubClusterLister:  hubClusterInformer.Lister(),
+		hubClusterClient:  hubClusterClient,
 		handleAcceptFalse: handleAcceptFalse,
 		recorder:          recorder,
 	}
 	return factory.New().
 		WithInformers(hubClusterInformer.Informer()).
+		ResyncEvery(3*time.Minute).
 		WithSync(c.sync).
 		ToController("HubAcceptController", recorder)
 }
 
 func (c *hubAcceptController) sync(ctx context.Context, _ factory.SyncContext) error {
-	cluster, err := c.hubClusterLister.Get(c.clusterName)
+	cluster, err := c.hubClusterClient.ClusterV1().ManagedClusters().Get(ctx, c.clusterName, metav1.GetOptions{})
 	if err != nil {
+		if apierrors.IsForbidden(err) {
```

Comment:
@qiujian16  Here, when rbac on hub side is removed, client will return forbidden error and captured by this controller then `handle` as accpet False.